src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Declarations.cs

```diff
@@ -11,37 +11,101 @@
 
 namespace Microsoft.CodeAnalysis.FindSymbols
 {
-    // Search query parameters.
-    internal struct SearchQuery
+    internal enum SearchKind
     {
-        // The predicate for matching names.  Never null.
-        public readonly Func<string, bool> Predicate;
+        /// <summary>
+        /// Use an case-sensitive comparison when searching for matching items.
+        /// </summary>
+        Exact,
+
+        /// <summary>
+        /// Use a case-insensitive comparison when searching for matching items.
+        /// </summary>
+        ExactIgnoreCase,
+
+        /// <summary>
+        /// Use a fuzzy comparison when searching for matching items. Fuzzy matching allows for 
+        /// a certain amount of misspellings, missing words, etc. See <see cref="SpellChecker"/> for 
+        /// more details.
+        /// </summary>
+        Fuzzy,
+
+        /// <summary>
+        /// Search term is matched in a custom manner (i.e. with a user provided predicate).
+        /// </summary>
+        Custom
+    }
 
-        // The name being searched for may be null in some cases.  But can be used for faster 
-        // index based searching if it is provided.
+    // Search query parameters.
+    internal class SearchQuery
+    {
+        // The name being searched for.  Is null in the case of custom predicate searching..  But 
+        // can be used for faster index based searching when it is available.
         public readonly string Name;
-        public readonly bool IgnoreCase;
 
-        public SearchQuery(string name, bool ignoreCase): 
-            this(n => ignoreCase ? CaseInsensitiveComparison.Comparer.Equals(name, n) : StringComparer.Ordinal.Equals(name, n))
+        // The kind of search this is.  Can be used for faster index based searching if it
```

Comment:
Suggestion: "Faster index-based searching can be used if the SearchKind is not Custom"